app/controllers/tiny_mce_assets_controller.rb

```diff
@@ -23,4 +28,88 @@ def create
       }, status: :unprocessable_entity
     end
   end
+
+  def download
+    asset = current_team.tiny_mce_assets.find_by_id(Base62.decode(params[:id]))
+    if asset&.image&.attached?
+      redirect_to rails_blob_path(asset.image, disposition: 'attachment')
+    else
+      render_404
+    end
+  end
+
+  def marvinjs_show
+    asset = current_team.tiny_mce_assets.find_by_id(Base62.decode(params[:id]))
+    return render_404 unless asset
+
+    render json: {
+      name: asset.image.metadata[:name],
+      description: asset.image.metadata[:description]
+    }
+  end
+
+  def marvinjs_create
+    result = MarvinJsService.create_sketch(marvin_params, current_user)
+    if result[:asset]
+      render json: {
+        image: {
+          url: rails_representation_url(result[:asset].preview),
+          token: Base62.encode(result[:asset].id),
+          source_type: result[:asset].image.metadata[:asset_type]
+        }
+      }, content_type: 'text/html'
+    else
+      render json: result[:asset].errors, status: :unprocessable_entity
+    end
+  end
+
+  def marvinjs_update
+    asset = MarvinJsService.update_sketch(marvin_params, current_user)
```

Comment:
So it basically allows to attach MJS asset to any object on the system?

Reply:
No, MJS connected to Asset or TinyMceAsset only.